lib/dal/src/component.rs

```diff
@@ -1921,13 +1975,29 @@ impl Component {
             Self::input_socket_attribute_values_for_component_id(ctx, to_component_id).await?;
         for (to_input_socket_id, input_socket_match) in input_sockets.into_iter() {
             if let Some(output_socket_match) =
-                Self::find_inferred_connection_to_input_socket(ctx, input_socket_match).await?
+                Self::find_potential_inferred_connection_to_input_socket(ctx, input_socket_match)
+                    .await?
             {
+                // add the check for to_delete on either to or from component
+                // Both "deleted" and not deleted Components can feed data into
+                // "deleted" Components. **ONLY** not deleted Components can feed
+                // data into not deleted Components.
+                let destination_component = Self::get_by_id(ctx, to_component_id).await?;
+                let source_component =
+                    Self::get_by_id(ctx, output_socket_match.component_id).await?;
+                let to_delete = !Self::should_data_flow_between_components(
```

Comment:
derives whether data should flow through based on the `to_delete` state of either end of the connection so we can handle these the same on the front end.